frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.unit.spec.tsx

```diff
@@ -0,0 +1,137 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import type { Engine } from "metabase-types/api";
+
+import DatabaseForm from "./DatabaseForm";
+
+interface SetupOpts {
+  onSubmit?: () => void;
+}
+
+function setup({ onSubmit }: SetupOpts) {
+  render(<DatabaseForm engines={engines} isAdvanced onSubmit={onSubmit} />);
+}
+
+describe("DatabaseForm", () => {
+  it("should submit default values", async () => {
+    const mockOnSubmit = jest.fn();
```

Comment:
I saw different setup in tests, when mocked function is returned from `setup` function with good name like `onSubmit`:

`const {onSubmit} = setup()`;

Reply:
good point. I agree that from a quick surf around our codebase I think creating and returning mock callbacks is more established.

Reply:
Use the suggested pattern